youtube_dl/extractor/yahoo.py

```diff
@@ -556,3 +560,153 @@ def _real_extract(self, url):
+                    'format_id': 'http-%s' % compat_str(vid.get('bitrate', '')),
+                    'height': int_or_none(vid.get('height')),
+                    'width': int_or_none(vid.get('width'))
+                })
+        self._remove_duplicate_formats(formats)
+        self._sort_formats(formats)
+
+        return formats
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        host = mobj.group('host')
+        # Headline pages without `displayid` default to `host`.
+        display_id = mobj.group('id') or host
+        webpage = self._download_webpage(url, display_id)
+
+        title = self._html_search_meta(
+            ['og:title', 'twitter:title'], webpage, 'title', default=None
+        ) or self._html_search_regex('<title>([^<]+)</title>', webpage, 'title')
+
+        if display_id == host:
+            # Headline page (w/ multiple BC playlists)
+            stream_urls_plists = re.findall(
+                r'YJNEWS\.LIVESTREAM\.SRC.+=(?:\s+)?["\'](?P<streamurl>[^"\']+plist=(?P<plist>\d+)[^"\']+)',
+                webpage)
+            if not stream_urls_plists:
+                # Manually build ('news.yahoo.co.jp', ...)
+                sbase_url = 'https://s.yimg.jp/images/media/video/player/html/embed_1.2.5.html?service=news&service_type=video&plist=%s&poster=%s&sp=%s&country=JP'
+
+                params = re.findall(
+                    r'["\']plist["\']:\s*["\'](?P<plist>[^"\']+)[^;]+?["\']poster["\']:\s*["\'](?P<poster>[^"\']+)[^;]+?["\']spaceId["\']:\s*["\'](?P<spaceid>[^"\']+)',
+                    webpage)
+                for p in params:
+                    stream_urls_plists.append((sbase_url % p, p[0]))
+
+            bc_url = 'http://players.brightcove.net/%s/%s_%s/index.html?playlistId=%s'
+            entries = []
+            for url, plist_id in stream_urls_plists:
+                embed_page = self._download_webpage(
+                    'https://s.yimg.jp/images/media/video/player/js/embed_1.0.9.min.js',
+                    plist_id,
+                    headers={
+                        'Authority': 's.yimg.jp',
+                        'Path': '/images/media/video/player/js/embed_1.0.9.min.js',
+                        'Referer': url,
+                    }, fatal=False)
+                account = self._search_regex(
+                    r'\w+\.video_attr\[(["\'])data-account\1\]\s*=\s*\1(?P<accountid>[\d]+)',
+                    embed_page, 'account id', group='accountid', fatal=False)
+                player = self._search_regex(
+                    r'\w+\.video_attr\[(["\'])data-player\1\]\s*=\s*\1(?P<playerid>[^"\']+)',
+                    embed_page, 'player id', group='playerid', fatal=False)
+                embed = self._search_regex(
+                    r'(["\'])data-embed\1\s*:\s*\1(?P<embed>[^"\']+)',
+                    embed_page, 'embed', group='embed', fatal=False)
+                if not all([embed_page, account, player, embed]):
```

Comment:
account, player and embed can be safely hardcoded instead of unnecessary download requests like in the rest of extractors. account is not likely to ever change.

Reply:
Good call. Done.